src/main/scala/is/hail/asm4s/FunctionBuilder.scala

```diff
@@ -117,17 +117,20 @@ class FunctionBuilder[F >: Null](parameterTypeInfo: Array[MaybeGenericTypeInfo[_
     ).emit(genericMn)
   }
 
-  def allocLocal[T]()(implicit tti: TypeInfo[T]): Int = {
+  def allocLocal[T](name: String = null)(implicit tti: TypeInfo[T]): Int = {
     val i = locals
     locals += tti.slots
 
     mn.localVariables.asInstanceOf[util.List[LocalVariableNode]]
-      .add(new LocalVariableNode("local" + i, tti.name, null, start, end, i))
+      .add(new LocalVariableNode(if (name == null) "local" + i else name, tti.name, null, start, end, i))
```

Comment:
I just want to double-check, local names are just informational, right?  Locals are always accessed by index.  Otherwise, the else should be something like `name + i` to ensure uniqueness.

Reply:
I can't find any authoritative information in the JVM spec about the name information that is stored with local variables in class files. The JVM spec 2.6.1 does state that local variables are "addressed by indexing".

I certainly only intend this to improve debug-ability of our generated byte code.